src/pd/async/util.rs

```diff
@@ -274,37 +281,32 @@ pub fn validate_endpoints(endpoints: &[String]) -> Result<(PDAsyncClient, GetMem
 
     match members {
         Some(members) => {
-            let (client, members) = try!(try_connect_leader(&members));
+            let (client, members) = try!(try_connect_leader(env.clone(), &members));
             info!("All PD endpoints are consistent: {:?}", endpoints);
             Ok((client, members))
         }
         _ => Err(box_err!("PD cluster failed to respond")),
     }
 }
 
-fn connect(addr: &str) -> Result<(PDAsyncClient, GetMembersResponse)> {
+fn connect(env: Arc<Environment>, addr: &str) -> Result<(PDClient, GetMembersResponse)> {
     debug!("connect to PD endpoint: {:?}", addr);
-    let ep = box_try!(Url::parse(addr));
-    let host = ep.host_str().unwrap();
-    let port = ep.port().unwrap();
-
-    let mut conf: grpc::client::GrpcClientConf = Default::default();
-    conf.http.no_delay = Some(true);
-
-    // TODO: It seems that `new` always return an Ok(_).
-    PDAsyncClient::new(host, port, false, conf)
-        .and_then(|client| {
-            // try request.
-            match client.GetMembers(GetMembersRequest::new()).wait() {
-                Ok(resp) => Ok((client, resp)),
-                Err(e) => Err(e),
-            }
-        })
-        .map_err(Error::Grpc)
+    let addr = if addr.starts_with("http://") {
+        &addr[7..]
```

Comment:
Try this: https://doc.rust-lang.org/std/string/struct.String.html#method.trim_left_matches